### ISSUE TITLE
Unable to read from a table that has Nullable Columns using the 'database/sql' interface

### ISSUE THREAD

**Issue by sankalp-amigo** [ClickHouse/clickhouse-go#754]
### Issue description
Unable to read Rows from a Table that has Null Values anywhere using the 'database/sql' interface. Even though 'sql.Nullstring' is being used. Expected to be able scan into a slice of 'sql.Nulstring' but instead get an error.

### Example code
```go
```
<img width="944" alt="image" src="https://user-images.githubusercontent.com/102962598/190998508-8e1bdb8f-e28e-4cd2-9009-c48c84ab06c8.png">

### Error log
```
If you have an error log, please paste it here.
```
<img width="944" alt="image" src="https://user-images.githubusercontent.com/102962598/190997996-e58bcb5f-0a06-48a7-b905-40d50d09177a.png">
sql: Scan error on column index 0, name "col1": unsupported Scan, storing driver.Value type *string into type *string

### Configuration
*OS:* Ubuntu 20.04.4 LTS

*Interface:* `database/sql`

*Driver version:* v2.3.0

*Go version:* go version go1.18.2 linux/amd64

*ClickHouse Server version:* version 22.9.1

**Comment by gingerwizard** [comment]
YOu need to pass **string not *string to represent the null value i.e. a pointer to a pointer.

**Comment by gingerwizard** [comment]
and https://github.com/ClickHouse/clickhouse-go/blob/main/tests/std/string_test.go#L60-L69

**Comment by gingerwizard** [comment]
Ok i can reproduce this. It will scan nil but not string values.

**Comment by gingerwizard** [comment]
we should support this. Workaround is use **string but NullString needs to work.

**Comment by gingerwizard** [comment]
This is subtly tricky. When using the std interface the method `Next` and is responsible for reading the rows - it requires the driver to implement `func (r *stdRows) Next(dest []driver.Value) error {`. Its here we actually read the value from the block. The dest slice gives us no indication as to whether the user intends to Scan into a NullString or string** - as the user later passes this in `Scan`.

We need to decide in Next whether to return a pointer or primitive string. For null we return a pointer - specifically [here](https://github.com/ClickHouse/clickhouse-go/blob/main/clickhouse_std.go#L294-L295). Why? because we support the user reading into a string**. 

Now at the point of Scan, the user passes string** for a Nullable(String). The sql.driver layer supports this marshaling and it works due [to this code](https://github.com/golang/go/blob/master/src/database/sql/convert.go#L401-L409). NullString needs a string value, however - and we have a pointer - see [1](https://github.com/golang/go/blob/master/src/database/sql/convert.go#L384-L386), then [2](https://github.com/golang/go/blob/master/src/database/sql/sql.go#L193). The string* to string* isn't supported inside sql.go i.e. [here](https://github.com/golang/go/blob/master/src/database/sql/convert.go#L219)

So in summary, by supporting string** we make supporting NullString tricky. Looking to see if we can do something in Scan.

**Comment by sankalp-amigo** [comment]
Using this workaround, I'm able to Read Nullable Strings but unable to read Nullable Integer or Float Columns that have any value (other than Null) in them. Which I could Read as String when the column was not Null and I was reading into *string.

```
sql: Scan error on column index 1, name \"fid\": unsupported Scan, storing driver.Value type *int32 into type *string"```

**Comment by gingerwizard** [comment]
Use **float and **intX - same principle.

**Comment by sankalp-amigo** [comment]
I do not have knowledge of the type of the columns of the table I'm querying.
I use strings because it is mentioned that numbers are auto converted to strings.
Which works fine for the case when none of my columns are Nullable. The Issue only arises when I have a **Nullable** Column of type NOT string and try to read it as string.

<img width="506" alt="image" src="https://user-images.githubusercontent.com/102962598/192086192-7085c49d-ee95-408f-a6b2-c6d95b7910b7.png">

**Comment by gingerwizard** [comment]
https://github.com/ClickHouse/clickhouse-go/blob/main/examples/std/dynamic_scan_types.go is an example where the column types are unknown.

I agree we need to address this, hence issue is open - I'm just not quite clear how to implement this. Open to PRs and/or ideas.

**Comment by stephenafamo** [comment]
Is there any update on this?

If supporting both `*string` and `sql.NullString` is impossible, I believe it is more in line with the database ecosystem to support `sql.NullString`

I'm not sure if there are any plans for a `v3` but that may be a reasonable time to introduce the breaking change.

**Comment by montanaflynn** [comment]
I was using `sql.Null*` and found out it's just inserting zero values which was surprising. I think a note in the docs should be added, I had to search through issues to find this behavior.
